src/django/api/facilities_download_view_set.py

```diff
@@ -1,94 +1,108 @@
 
 
-class FacilitiesDownloadViewSet(mixins.ListModelMixin,
-                                viewsets.GenericViewSet):
+class FacilitiesDownloadViewSet(
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet
+):
     """
     Get facilities in array format, suitable for CSV/XLSX download.
     """
     queryset = FacilityIndex.objects.all()
-    pagination_class = PageAndSizePagination
+    pagination_class = None
 
-    def get_serializer(self, page_queryset):
+    def __is_embed_mode(self):
+        return self.request.query_params.get('embed') == '1'
+
+    def get_serializer(self, objs):
         if self.__is_embed_mode():
             contributor_id = get_embed_contributor_id_from_query_params(
                 self.request.query_params
             )
             return FacilityDownloadSerializerEmbedMode(
-                page_queryset, many=True, contributor_id=contributor_id
+                objs,
+                many=True,
+                contributor_id=contributor_id
             )
-
-        return FacilityDownloadSerializer(page_queryset, many=True)
-
-    def __is_embed_mode(self):
-        return self.request.query_params.get('embed') == '1'
+        return FacilityDownloadSerializer(objs, many=True)
 
     def list(self, request):
-        """
-        Returns a list of facilities in array format for a given query.
-        (Maximum of 250 facilities per page.)
-        """
         FacilitiesDownloadService.check_if_downloads_are_blocked()
         FacilitiesDownloadService.validate_query_params(request)
         FacilitiesDownloadService.log_request(request)
 
-        queryset = FacilitiesDownloadService.get_filtered_queryset(request)
-        total_records = queryset.count()
-        facility_download_limit = None
+        base_qs = FacilitiesDownloadService.get_filtered_queryset(request)
 
+        limit = None
         if (
             not switch_is_active('private_instance')
             and not self.__is_embed_mode()
         ):
-            facility_download_limit = FacilitiesDownloadService \
-                .get_download_limit(request)
+            limit = FacilitiesDownloadService.get_download_limit(request)
+
+        page = int(request.query_params.get('page', 1) or 1)
+        page_size = int(request.query_params.get('pageSize', 250) or 250)
+        is_first_page = (page == 1)
+
         FacilitiesDownloadService.enforce_limits(
-            request,
-            total_records,
-            facility_download_limit
+            qs=base_qs, limit=limit, is_first_page=is_first_page
         )
 
-        page_queryset = FacilitiesDownloadService \
-            .check_pagination(self.paginate_queryset(queryset))
-        list_serializer = self.get_serializer(page_queryset)
+        items, is_last_page = FacilitiesDownloadService.\
+            fetch_page_and_cache(
+                base_qs,
+                request,
+                page,
+                page_size,
+                block=50
```

Comment:
It looks like a magic number. What about moving this to constants related to pagination like:
```
class PaginationConfig:
    DEFAULT_BLOCK_SIZE = 50
    CACHE_TTL_SECONDS = 1800
    MAX_PAGE_SIZE = 1000
```